src/safe/encoder.rs

```diff
@@ -261,6 +251,14 @@ impl NV_ENC_PIC_PARAMS {
         }
     }
 
+    #[must_use]
+    pub fn end_of_stream() -> Self {
+        NV_ENC_PIC_PARAMS {
+            encodePicFlags: NV_ENC_PIC_FLAGS::NV_ENC_PIC_FLAG_EOS as u32,
+            ..Default::default()
+        }
+    }
```

Comment:
See https://docs.nvidia.com/video-technologies/video-codec-sdk/11.1/nvenc-video-encoder-api-prog-guide/index.html#notifying-the-end-of-input-stream, setting the flag and struct version should be enough if I understand correctly,

Reply:
and yes, I have tested it. The output is the same apart from that this actually has 128 frames instead of 129 in the original version.

Reply:
> No input buffer is required while calling NvEncEncodePicture for EOS notification. 
Great, so no input buffer is required. What about the output bitstream though?

Reply:
> EOS notification effectively flushes the encoder. This can be called multiple times in a single encode session. This operation however must be done before closing the encode session.

Sounds to me like it doesn't output anything. Also, the fact that he documentation says:
> and all other members of `NV_ENC_PIC_PARAMS` set to 0

would imply to me that the input should also be set to null pointers where possible.